contracts/modules/PermissionManager/GeneralPermissionManager.sol

```diff
@@ -27,72 +32,181 @@ contract GeneralPermissionManager is IPermissionManager, Module {
-        emit AddPermission(_delegate, _details, now);
+        allDelegates.push(_delegate);
+        emit AddDelegate(_delegate, _details, now);
     }
 
-  /**
-    * @notice Use to provide/change the permission to the delegate corresponds to the module contract
-    * @param _delegate Ethereum address of the delegate
-    * @param _module Ethereum contract address of the module
-    * @param _perm Permission flag
-    * @param _valid Bool flag use to switch on/off the permission
-    * @return bool
-    */
+    /**
+     * @notice use to check if an address is a delegate or not
+     * @param _potentialDelegate the address of potential delegate
+     * @return bool
+     */
+    function checkDelegate(address _potentialDelegate) external view returns(bool) {
+        require(_potentialDelegate != address(0));
+
+        if (delegateDetails[_potentialDelegate] != bytes32(0)) {
+            return true;
+        } else
+            return false;
+    }
+
+    /**
+     * @notice Use to provide/change the permission to the delegate corresponds to the module contract
+     * @param _delegate Ethereum address of the delegate
+     * @param _module Ethereum contract address of the module
+     * @param _perm Permission flag
+     * @param _valid Bool flag use to switch on/off the permission
+     * @return bool
+     */
     function changePermission(
         address _delegate,
         address _module,
         bytes32 _perm,
         bool _valid
     )
-    external
+    public
     withPerm(CHANGE_PERMISSION)
-    returns(bool)
     {
-        require(delegateDetails[_delegate] != bytes32(0), "Delegate details not set");
+        require(delegateDetails[_delegate] != bytes32(0), "Delegate is not exists");
         perms[_module][_delegate][_perm] = _valid;
         emit ChangePermission(_delegate, _module, _perm, _valid, now);
-        return true;
     }
 
     /**
-    * @notice Use to get the details of the delegate
-    * @param _delegate Ethereum address of the delegate
-    * @return Details of the delegate
-    */
-    function getDelegateDetails(address _delegate) external view returns(bytes32) {
-        return delegateDetails[_delegate];
+     * @notice Use to change one or more permissions for a single delegate at once
+     * @param _delegate Ethereum address of the delegate
+     * @param _modules Multiple module matching the multiperms, needs to be same length
+     * @param _perms Multiple permission flag needs to be changed
+     * @return nothing
+     */
+    function changePermissionMulti(
+        address _delegate,
+        address[] _modules,
+        bytes32[] _perms
+    )
+    external
+    withPerm(CHANGE_PERMISSION)
+    {
+        require(_modules.length > 0 && _perms.length > 0, "0 length is not allowed");
+        require(_modules.length == _perms.length, "Array length mismatch");
+        
+        for(uint8 i = 0; i < _perms.length; i++) {
+            bool _perm = !perms[_modules[i]][_delegate][_perms[i]];
```

Comment:
Hi @comeonbuddy - to make the behaviour here more explicit please could you take in a bool[] and set the permissions to match it, rather than flipping the permission,